{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create a transcirptome using reference genome and annotated sequences\n",
    "gffread -w csi.transcriptome.fa -g csi.chromsome.fa stringtie_merged.gtf\n",
    "\n",
    "#create the index from the transcriptome that was created\n",
    "salmon index -t csi.transcriptome.fa -i csi.transcriptome_index\n",
    "\n",
    "#script for quantifying all of the samples downloaded from the citrus genome project\n",
    "#!/bin/bash\n",
    "\n",
    "cd /data/VALENCIA\n",
    "# create a list so that a pattern can be used to run a script\n",
    "ls *_1.fq.gz | cut -d'_'-f 1 > fastq.list\n",
    "\n",
    "#iterate the quantification process for each sample\n",
    "for file in fastq.list;\n",
    "do\n",
    "echo \"Processing sample ${file}\"\n",
    "salmon quant -i /data/Salmon/csi.transcriptome_index -l A \\\n",
    "    -1 ${file}_1.fq.gz \\\n",
    "    -2 ${file}_2.fq.gz \\ \n",
    "    -p 40 --validateMappings -o /data/Salmon/quants/${file}_quant\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import the transcripts from Salmon for gene analysis w/ DESeq2\n",
    "# created a file path\n",
    "for (file in f){dir <- paste0(\"/RNA/quants/\", f, \"_quant)}\n",
    "\n",
    "# read in the table with sample OD's which will be combined later with the transcript ocounts\n",
    "samples <- read.csv(\"VALENCIA/phenotype.csv\", header = TRUE)\n",
    "\n",
    "# changed the Conditions columns to numeric values in order to comply w packages\n",
    "samples[\"Condition\"] <- c('1','1','1','2','2','2','3','3','3','4','4','4','5','5','5')\n",
    "\n",
    "# create txt files to aling the sample ID and counts\n",
    " echo -e \"TXNAME\\tGENEID\" > tx2gene.txt; awk '$3 == \"transcript\" {print $12\"\\t\"$10}' stringtie_merged.gtf | sed 's/\"//g;s/;//g' >> tx2gene.txt\n",
    "\n",
    "# read in the txt file of the tscript and gene ID\n",
    "tx2gene <- read.table(\"VALENCIA/tx2gene.txt\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# import quant/sf files by creating a vector with the file names\n",
    "\n",
    "#files <- file.path(dir,\"salmon\", samples$sample, \"quant.sf.gz\")\n",
    "files <- file.path(dir, \"/quant.sf\")\n",
    "\n",
    "# use the df from earlier to create a list with all the file names\n",
    "f <- samples[,1]\n",
    "# rename the samples in \"files\" \n",
    "for(file in files){names(files) <- paste0(f)}\n",
    "\n",
    "# import the quant.sf files\n",
    "txi.salmon <- tximport(files, type = \"salmon\", tx2gene=tx2gene)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# create a DESeq data set\n",
    "# create a df to align the samples with their condition\n",
    "sampleTable <- data.frame(condition = factor(rep(c('1','2','3','4','5'), each = 3)))\n",
    "rownames(sampleTable) <- colnames(txi.salmon$counts)\n",
    "# create a DESeq data set with the sample counts ready for export\n",
    "dds <-DESeqDataSetFromTximport(txi.salmon, sampleTable, ~condition)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run DESeq2 on the dataset \n",
    "dds1 <- DESeq(dds)\n",
    "\n",
    "#build a table for each of the results you want to analyze\n",
    "res_GHA_PFA <- results(dds1, contrast=c(\"condition\",\"2\",\"1\"))\n",
    "res_102_103 <- results(dds1, contrast=c(\"condition\",\"2\",\"3\"))\n",
    "res_102_108 <- results(dds1, contrast=c(\"condition\",\"2\",\"4\"))\n",
    "res_102_PV5 <- results(dds1, contrast=c(\"condition\",\"2\",\"5\"))\n",
    "res_103_108 <- results(dds1, contrast=c(\"condition\",\"3\",\"4\"))\n",
    "res_103_PV5 <- results(dds1, contrast=c(\"condition\",\"3\",\"5\"))\n",
    "res_108_PV5 <- results(dds1, contrast=c(\"condition\",\"4\",\"5\"))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Save the results to a csv file\n",
    "# create a list of strings to save the df to\n",
    "res <- c('res_GHA_PFA', 'res_102_103', 'res_102_108', 'res_102_PV5', 'res_103_108', 'res_103_PV5', 'res_108_PV5')\n",
    "\n",
    "# create a list to iterate the dfs\n",
    "res.list <- list(res_GHA_PFA, res_102_103, res_102_108, res_102_PV5, res_103_108, res_103_PV5, res_108_PV5)\n",
    "\n",
    "# Order each of the lists by padj\n",
    "lapply(res.list, function(x) res.list_sort <- x[order(x$padj),])\n",
    "\n",
    "# write the dataframe to csv\n",
    "for (i in 1:7){write.csv(res.list_order[i], paste(\"/RNA/VALENCIA/\", res[i], \".csv\", sep=\" \"), row.names = TRUE)}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create graphs for each of the dataframes\n",
    "lapply(res.list, function(x) plotMA(x, ylim=c(-10,10)))\n",
    "\n",
    "#save the graphs\n",
    "for(i in 1:7){\n",
    "    svg(paste0(\"/RNA/quants/\", res[i], \".svg\"), width = 7, height = 7)\n",
    "    lapply(res.list, function(x) plotMA(x, ylim=c(-10,10)))\n",
    "    \n",
    "dev.off()\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compare significant genes\n",
    "#subset all the padj values and place them into one list\n",
    "DEG <- lapply(res.list, subset, padj <0.05)\n",
    "#concatenate all of the significant genes\n",
    "do.call(rbind, DEG)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compare the significant genes\n",
    "colnames(comp) <- c(\"comp1\")\n",
    "# create an list of all the gene names\n",
    "res_names.list <- lapply(res.list, function(x) rownames(x))\n",
    "\n",
    "#remove duplicated genes\n",
    "DEG <- unique(DEG)\n",
    "\n",
    "# create a vector of gene names\n",
    "genes <- as.vector(rownames(DEG))\n",
    "\n",
    "# compare the genes\n",
    "sapply(genes, function(x) x %in% res.list[i])\n",
    "\n",
    "# iterate this function and output to the list\n",
    "comp <- data.frame()\n",
    "for (i in 1:7){comp[i] <- cbind(lapply(genes, function(x) x %in% res_names.list[i])}\n",
    "\n",
    "# change logical values to numeric\n",
    "as.numeric(comp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# subset all the pval values \n",
    "PVAL <- lapply(res.list, subset, pvalue<0.05)\n",
    "\n",
    "#concatenate the values\n",
    "PVAL.list<- do.call(rbind, PVAL)\n",
    "\n",
    "# output names as a vector\n",
    "PVA_names <- lapply(PVAL, function(x) rownames(x))\n",
    "\n",
    "# output gene names as a vector\n",
    "genes1 <- as.vector(rownames(PVAL.list))\n",
    "\n",
    "#remove duplicated genes\n",
    "genes1 <- unique(genes1)\n",
    "\n",
    "# create a deg column for the dataframe\n",
    "comp1 <- data.frame(genes1)\n",
    "\n",
    "#iterate an function to match the genes\n",
    "# unlist PVA object to create a vector so is.element can be used\n",
    "for (i in 2:8){comp1[i] <- is.element(genes1,unlist(PVA_names[i-1]))}\n",
    "\n",
    "#convert data to numeric\n",
    "cols <- sapply(comp1, is.logical)\n",
    "comp1[,cols] <- lapply(comp1[,cols], as.numeric)\n",
    "\n",
    "# find the total matched genes of the data. convert data to isnumeric\n",
    "sum(comp1[i]) #sum of a column in the dataframe\n",
    "Total<-colSums(comp1[2:8])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "install.packages('VennDiagram')\n",
    "#create a venn diagram\n",
    "# change the column names\n",
    "colnames(comp1)<- c('geneID','GHA_PFA','102_103','102_108','102_PV5','103_108','103_PV5','108_PV5')\n",
    "\n",
    "# create groups to put into venn diagram\n",
    "col102 <- c(3:5)\n",
    "col103 <- c(3,6,7)\n",
    "colPV5 <- c(5,7,8)\n",
    "# create multiple triple venn diagrams\n",
    "venn(comp1[,102], snames = c('102-102','102-108','102-PV5'),counts = 1, zcolor = \"style\")\n",
    "venn(comp1[,103], snames = c('102-102','102-108','102-PV5'),counts = 1, zcolor = \"style\")\n",
    "venn(comp1[,PV5], snames = c('102-102','102-108','102-PV5'),counts = 1, zcolor = \"style\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "lapply(PVAL, function(x) PVAL_order <- x[order(x$pvalue),])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Separate upregulated DEG and downregulated DEG\n",
    "# check to see if the logfold2change provided by pipeline is which ratio\n",
    "# retrieve normalized values of each sample\n",
    "dds <- estimateSizeFactors(dds)\n",
    "counts(dds, normalized=TRUE)\n",
    "\n",
    "#find sum each condition for  a gene\n",
    "Cs1g01170 <- head(counts(dds, normalized=TRUE), n=1)\n",
    "\n",
    "sum(Cs1g01170[1:3])\n",
    "sum(Cs1g01170[4:6])\n",
    "sum(Cs1g01170[7:9])\n",
    "sum(Cs1g01170[10:12])\n",
    "sum(Cs1g01170[13:15])\n",
    "\n",
    "# compared the values with the ones received from the pipeline, which matched\n",
    "# subset the upregulated 102 genes at p < 0.05 FC >= 1\n",
    "upreg102 <- lapply(PVAL[2:4], subset, log2FoldChange >=1)\n",
    "#subset the downregulated 102 genes at p <0.05 FC <= -1\n",
    "dowreg102 <- lapply(PVAL[2:4], subset, log2FoldChange <=-1)\n",
    "\n",
    "#subset upregulated 103 genes \n",
    "upreg103 <- lapply(PVAL[2], subset, log2FoldChange <=-1)\n",
    "upreg103[2:3] <- lapply(PVAL[5:6], subset, log2FoldChange >=1)\n",
    "#subset downregulated 103 genes\n",
    "downreg103 <- lapply(PVAL[2], subset, log2FoldChange >=1)\n",
    "downreg103[2:3] <- lapply(PVAL[5:6], subset, log2FoldChange <=-1)\n",
    "\n",
    "#subset upregulated 108 genes\n",
    "upreg108 <- lapply(PVAL[3], subset, log2FoldChange <= -1)\n",
    "upreg108[2] <- lapply(PVAL[5], subset, log2FoldChange <= -1)\n",
    "upreg108[3] <- lapply(PVAL[7], subset, log2FoldChange >= 1)\n",
    "#subset downregulated 108 genes\n",
    "downreg108 <- lapply(PVAL[3], subset, log2FoldChange >= 1)\n",
    "downreg108[2] <- lapply(PVAL[5], subset, log2FoldChange >= 1)\n",
    "downreg108[3] <- lapply(PVAL[7], subset, log2FoldChange <= -1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create a csv  of all the regulated expressed genes\n",
    "# 1) sort the original file of genes alphanumerically by gene ID\n",
    "# read res.list as a data frame\n",
    "as.data.frame(res.list[i])\n",
    "# read the rownames of the data frame\n",
    "head(rownames(as.data.frame(res.list[i])))\n",
    "# order the geneID \n",
    "install.packages('gtools')\n",
    "library(gtools)\n",
    "mixedorder(rownames(as.data.frame(res.list[i])))\n",
    "\n",
    "# 2) isolate the values in reg.list \n",
    "head(subset(as.data.frame(res.list[i]), select = c('log2FoldChange','pvalue')))\n",
    "# apply this to each list in res.list\n",
    "l<-(lapply(res.list, function(x) (subset(as.data.frame(x),select = c('log2FoldChange','pvalue')))))\n",
    "\n",
    "\n",
    "# 3) output to an empty dataframe concatenated. geneID|log2FoldChange[i]|pvalue[i]|...\n",
    "reggenes <- cbind(as.data.frame(l[1:7]))\n",
    "# change the column names so that they're easier to manage\n",
    "colnames(reggenes) <- c('log2FC1','pval1','log2FC2','pval2','log2FC3','pval3','log2FC4','pval4','log2FC5','pval5','log2FC6','pval6','log2FC7','pval7')\n",
    "\n",
    "\n",
    "#4) subset the up and down regulated items\n",
    "# subset them to have pvalues less than 0.05\n",
    "head(subset(reggenes, (pval1 < 0.05)))\n",
    "head(subset(reggenes, (pval2 < 0.05)))\n",
    "\n",
    "#5) subset them to which log2FC values are <= -1 for sample 2 and >= 1 for sample 5 and 6\n",
    "# use logical conjunction\n",
    "subset(reggenes, (log2FC2 <=-1 & pval2 <0.05) & (log2FC5 >=1 & pval5 < 0.05) & (log2FC6 >=1 & pval6 <0.05), select= c('log2FC2','pval2','log2FC5','pval5','log2FC6','pval6'))\n",
    "upregulated <- subset(reggenes, (log2FC2 <=-1 & pval2 <0.05) & (log2FC5 >=1 & pval5 < 0.05) & (log2FC6 >=1 & pval6 <0.05), select= c('log2FC2','pval2','log2FC5','pval5','log2FC6','pval6'))\n",
    "#change column names\n",
    "colnames(upregulated) <- c('102log2FC','102pval','108log2FC','108pval','PV5log2FC','PV5pval')\n",
    "#subset downregulated\n",
    "subset(reggenes, (log2FC2 >=1 & pval2 <0.05) & (log2FC5 <=-1 & pval5 < 0.05) & (log2FC6 <=-1 & pval6 <0.05), select= c('log2FC2','pval2','log2FC5','pval5','log2FC6','pval6'))\n",
    "downregulated <- subset(reggenes, (log2FC2 >=-1 & pval2) & (log2FC5 <=-1 & pval5) & (log2FC6 <=-1 & pval6), select= c('log2FC2','pval2','log2FC5','pval5','log2FC6','pval6'))\n",
    "colnames(downregulated) <- c('102log2FC','102pval','108log2FC','108pval','PV5log2FC','PV5pval')\n",
    "\n",
    "#5a) output to csv\n",
    "write.csv(upregulated, file = \"/RNA/quants/103upreg.csv\")\n",
    "write.csv(downregulated, file = \"/RNA/quants/103downreg.csv\")\n",
    "write.csv(reggenes, file= \"/RNA/quants/ordergenes\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#6) subset the upregulated and downregulated genes of the other comparisons\n",
    "upreggha_pfa<-subset(reggenes, (log2FC1 >= 1 & pval1 < 0.05), select = c('log2FC1', 'pval1'))\n",
    "downreggha_pfa<-subset(reggenes, (log2FC1 <=-1 & pval1 < 0.05), select = c('log2FC1', 'pval1'))\n",
    "\n",
    "upreg102_103<-subset(reggenes, (log2FC2 >= 1 & pval2 < 0.05), select = c('log2FC2', 'pval2'))\n",
    "downreg102_103<-subset(reggenes, (log2FC2 <=-1 & pval2 < 0.05), select = c('log2FC2', 'pval2'))\n",
    "\n",
    "upreg102_108<-subset(reggenes, (log2FC3 >= 1 & pval3 < 0.05), select = c('log2FC3', 'pval3'))\n",
    "downreg102_108<-subset(reggenes, (log2FC3 <=-1 & pval3 < 0.05), select = c('log2FC3', 'pval3'))\n",
    "\n",
    "upreg102_PV5<-subset(reggenes, (log2FC4 >= 1 & pval4 < 0.05), select = c('log2FC4', 'pval4'))\n",
    "downreg102_PV5<-subset(reggenes, (log2FC4 <=-1 & pval4 < 0.05), select = c('log2FC4', 'pval4'))\n",
    "\n",
    "upreg103_108<-subset(reggenes, (log2FC5 >= 1 & pval5 < 0.05), select = c('log2FC5', 'pval5'))\n",
    "downreg103_108<-subset(reggenes, (log2FC5 <=-1 & pval5 < 0.05), select = c('log2FC5', 'pval5'))\n",
    "\n",
    "upreg103_PV5<-subset(reggenes, (log2FC6 >= 1 & pval6 < 0.05), select = c('log2FC6', 'pval6'))\n",
    "downreg103_PV5<-subset(reggenes, (log2FC6 <=-1 & pval6 < 0.05), select = c('log2FC6', 'pval6'))\n",
    "\n",
    "upreg108_PV5<-subset(reggenes, (log2FC7 >= 1 & pval7 < 0.05), select = c('log2FC7', 'pval7'))\n",
    "downreg108_PV5<-subset(reggenes, (log2FC7 <=-1 & pval7 < 0.05), select = c('log2FC7', 'pval7'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#output the data to csv files\n",
    "write.csv(upreggha_pfa, file = \"/RNA/quants/upregGHA_PFA.csv\")\n",
    "write.csv(downreggha_pfa, file = \"/RNA/quants/downregGHA_PFA.csv\")\n",
    "\n",
    "write.csv(upreg102_103, file = \"/RNA/quants/upreg102_103.csv\")\n",
    "write.csv(downreg102_103, file = \"/RNA/quants/downreg102_103.csv\")\n",
    "\n",
    "write.csv(upreg102_108, file = \"/RNA/quants/upreg102_108.csv\")\n",
    "write.csv(downreg102_108, file = \"/RNA/quants/downreg102_108.csv\")\n",
    "\n",
    "write.csv(upreg102_PV5, file = \"/RNA/quants/upreg102_PV5.csv\")\n",
    "write.csv(downreg102_PV5, file = \"/RNA/quants/downreg102_PV5.csv\")\n",
    "\n",
    "write.csv(upreg103_108, file = \"/RNA/quants/upreg103_108.csv\")\n",
    "write.csv(downreg103_108, file = \"/RNA/quants/downreg103_108.csv\")\n",
    "\n",
    "write.csv(upreg103_PV5, file = \"/RNA/quants/upreg103_PV5.csv\")\n",
    "write.csv(downreg103_PV5, file = \"/RNA/quants/downreg103_PV5.csv\")\n",
    "\n",
    "write.csv(upreg108_PV5, file = \"/RNA/quants/upreg108_PV5.csv\")\n",
    "write.csv(downreg108_PV5, file = \"/RNA/quants/downreg108_PV5.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
